build/xwalk.gni

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2016 Intel Corporation. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# This file contains Crosswalk-feature-related build flags. Users should
+# import it into their build folder by run:
+#   echo 'import("//xwalk/build/xwalk.gni")' > out/your_folder/args.gn'
+# Then run:
+#   gn gen out/your_folder
+
+# Set it false when building content_shell.
+declare_args() {
```

Comment:
Finally I found what causes the mis-understanding:
I assume when running `"gn gen"` for crosswalk,  all the upstream original targets in `src/BUILD.gn` (like `group("ALL")`) should never be triggered,  no ninja files should be generated for those upstream targets.  Only those targets DEPsed by xwalk will be generated into ninja files.
In the approach of `"build/secondary"` or `"root_extra_deps"`,  the `group("All")` will be generated into ninja files although they are not compiled.
I guess maybe it's why you meet the error of 
```
  "Collisions:
    //chrome:repack_locales_pack_en-US
    //xwalk/resources:xwalk_strings"
```
It is because the chrome targets conflict with crosswalk targets,  they both try to output `locales/en-US.pak` which is used for locales in resource_bundle module.